src/main/java/net/dv8tion/jda/core/audio/AudioNatives.java

```diff
@@ -22,12 +22,24 @@
 
 import java.io.IOException;
 
-public class AudioNatives
+public final class AudioNatives
 {
     private static final Logger LOG = LoggerFactory.getLogger(AudioNatives.class);
     private static boolean initialized;
     private static boolean audioSupported;
 
+    private AudioNatives() {}
+
+    public static synchronized boolean loadFrom(String absolutePath)
+    {
+        if (initialized)
+            return false;
```

Comment:
should this return `audioSupported?`

Reply:
`ensureOpus()` returns `audioSupported` because it is used by JDA to check whether the library is available or not. `loadFrom()` returns true if the library is loaded from that method invokation and false if not (it was already loaded).

```java
if (AudioNatives.loadFrom("/lib/.../libopus.so")) {
    log.info("Loaded custom library at /lib/.../libopus.so");
} else {
    log.info("Opus binaries were already loaded, skipping.");
}
```